### ISSUE TITLE
Manually selecting release and mapping doesn't carry over to import queue and requires manual import also

### ISSUE THREAD

**Issue by ZerkerEOD** [Sonarr/Sonarr#6147]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

When manually importing and selecting the episode mapping, it doesn't carry over to auto import and doesn't keep the episode mapping.

### Expected Behavior

When manually searching and selecting the episode mapping, it should be able to auto-import with the mapping selected during the initial download as it should be assumed that it was already reviewed.

### Steps To Reproduce

When I perform a manual search for an episode that has known issues such as American Dad, and find the episode I want based off of the name. 

![image](https://github.com/Sonarr/Sonarr/assets/26573385/b2c5abef-3d27-4700-b95d-31e6961eb17e)

Then I click the override and add to queue button, below is what is portrayed even though asking for s3e11 it defaults to s4e11. Now that isn't as much of a problem but I think if doing manual search it should default to the episode I am looking for as I am selecting it. 
 
![image](https://github.com/Sonarr/Sonarr/assets/26573385/78368a1b-471a-4fb0-a573-93bbfccc6ab9)

Now comes what I think is the bug, If I edit it telling it no, I want it mapped to s3e11, then I would think that I have already told it it's correct mapping and it should be done.

![image](https://github.com/Sonarr/Sonarr/assets/26573385/e9057c85-1fb2-4103-85d6-a9531091581c)

So, I grab the release thinking it is all good, but it isn't. Now, I have to go back into the queue and see it didn't import.

 
![image](https://github.com/Sonarr/Sonarr/assets/26573385/b408c290-5978-44da-808d-e426bd035c0f)

Then I have to manually import it and select the season and episode again.

### Environment

```markdown
- OS: Unraid
- Sonarr: 4.0.0.688
- Docker Install: Yes
- Using Reverse Proxy: Yes, but in this instance I am directly connected to the instance with no proxy
- Browser: Brave/Chrome
- Database: Sqlite 3.41.2
```

### What branch are you running?

Develop

### Trace Logs?

The reproduction with the trace logs are for American Dad S3E12 named A.T. The Abusive Terrestrial with gets mapped to S4E12 as it looks like there is way more in the log and would help identify the path that the specific episode is on.

https://pastebin.com/wq0tC1SD

### Anything else?

I think I managed to capture what I am seeing and what I was expecting above and I hope it helps.

**Comment by bakerboy448** [comment]
If SQ is following TVDb not xem submit a mapping request - https://wiki.servarr.com/sonarr/faq#problematic-shows

**Comment by ZerkerEOD** [comment]
@bakerboy448, it is already on the request and it is a known bad which I understand. I think the bug is when I manually search and select the release that I want it fails to auto import with what I selected which was already reviewed manually before the request. I don't think the mapping would work either way as the episodes are all messed up which is why I am manually looking for ones with the episode title included so I can verify it isn't for a different mapped episode.

**Comment by stevietv** [comment]
Indeed, the Overrride and Grab is not sticky from grab to import, This is currently intentional, as Sonarr parses release name on grab and filename on import, and thus making it not stick ensures that there isn't an incorrect file in the download.

**Comment by ZerkerEOD** [comment]
Am I missing something then, a lot of the episodes I download turn out to be an encoded file name like such. 
![image](https://github.com/Sonarr/Sonarr/assets/26573385/1f99874f-e80f-45e1-a552-38031e2abed0)

**Comment by ZerkerEOD** [comment]
Update, I found an option to deobfuscate file names and checked it. @stevietv, the import still does not appear to be working off the file name though 
![image](https://github.com/Sonarr/Sonarr/assets/26573385/7a001d06-9977-441e-967e-77394dd13a3b)

**Comment by stevietv** [comment]
Sonarr is parsing again S07E01 to be Season 8 Episode one - in this case due to TheXEM,info mappings present for this show.
No matter what you set as override on grab sonarr will parse the filename again before importing

**Comment by markus101** [comment]
This is quite a complex problem because there is no guarantee that there is a 1:1 mapping between files in the release and the release name, regardless of what you've selected. A multi-episode release could be 1 file or it could be 1 for each file, a season pack may be incomplete or poorly named so Sonarr can't discern what is what.

We can likely improve this scenario for a small subset of manually mapped and grabbed releases, but we'll need to make sure it doesn't extend beyond single file releases and potentially only when a single episode is grabbed, but definitely not season packs.

**Comment by ZerkerEOD** [comment]
@markus101, I agree that it shouldn't work for season packs, that could cause a ton of issues. I'm only talking about when manually searching and selecting an episode to download, if it's a season packs I would prefer it to fail and need to be manually imported as well.

**Comment by nirurin** [comment]
I'm curious then... what's the point in the 'override and grab' option even existing? Why allow people to manually set a season and episode target for the download, if that information is then ...... completely ignored by every other stage of the process? Seems like a waste of time.

**Comment by Xevion** [comment]
Currently mystified by this right now. What does this button even do? The episode/season being parsed out of it is completely incorrect. I thought override would work. It effectively does nothing to my knowledge. How am I supposed to force a download? Manual import isn't even possible as it errors somehow before I get the option to do a manual import. Baffled right now.

**Comment by bakerboy448** [comment]
Please use one of the support channels: [forums](https://forums.sonarr.tv/), [discord ](https://discord.gg/Ex7FmFK), or [IRC ](https://web.libera.chat/?channels=#sonarr)for support/questions.

Override has nothing to do with manual import and manual import has nothing to do with grabbing a release and sending to your download client which sounds like is failing.

**Comment by rollsveallambchopfig** [comment]
> Update, I found an option to deobfuscate file names and checked it. [@stevietv](https://github.com/stevietv), the import still does not appear to be working off the file name though ![image](https://github.com/Sonarr/Sonarr/assets/26573385/7a001d06-9977-441e-967e-77394dd13a3b)

I'm curious to know what you found to deobfuscate file names as I have this issue semi-frequently too.
